core/src/bmi.jl

```diff
@@ -62,7 +62,7 @@ function BMI.initialize(T::Type{Model}, config::Config)::Model
     timespan = (zero(t_end), t_end)
 
     jac_prototype = get_jac_prototype(parameters)
-    RHS = ODEFunction(water_balance!; jac_prototype)
+    RHS = ODEFunction(water_balance!; jac_prototype, jac = water_balance_jac!)
```

Comment:
Just removing the `jac_prototype` as an argument fails because the `water_balance_jac!` expects a sparse matrix. In `jac.jl` replacing all `J::SparseMatrixCSC{Float64, Int64}` by `J::Union{SparseMatrixCSC{Float64, Int64}, Matrix{Float64}}` makes it work again but then the computation is probably not done sparsely anymore.

The sparsity structure can probably not reliably be inferred from a call to `water_balance_jac!` anyway since some nodes sometimes but not always contribute to the Jacobian. Maybe this could work if it is not a problem that the sparsity structure has to be updated during simulation.

Reply:
Good to know, thanks for checking!